backend/services/product/internal/product/product.go

```diff
@@ -0,0 +1,116 @@
+package product
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/manaaan/ekolivs-oms/pkg/env"
+	"github.com/manaaan/ekolivs-oms/pkg/zettle"
+	"github.com/manaaan/ekolivs-oms/product/api"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type Service struct {
+	zettleService *zettle.Service
+	storeService  *Store
+}
+
+func New() (*Service, error) {
+	zettleService, err := zettle.New(zettle.ServiceNewParams{
+		ClientId: env.Required("ZETTLE_ORG_UUID"),
+		ApiKey:   env.Required("ZETTLE_API_KEY"),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &Service{
+		zettleService: zettleService,
+		storeService:  &Store{},
+	}, nil
+}
+
+func (s Service) GetProducts() ([]*api.Product, error) {
+	zettleProducts, err := s.zettleService.GetProducts()
+	if err != nil {
+		return nil, err
+	}
+
+	storeProducts, err := s.storeService.GetProducts()
+	fmt.Println(storeProducts)
+
+	products := []*api.Product{}
+	for _, zettleProduct := range zettleProducts {
+		for _, variant := range zettleProduct.Variants {
+			var imageUrl *string
+			if zettleProduct.Presentation != nil {
+				imageUrl = zettleProduct.Presentation.ImageUrl
+			}
+
+			products = append(products, &api.Product{
+				Id:            variant.Uuid.String(),
+				Name:          zettleProduct.Name,
+				Sku:           variant.Sku,
+				Barcode:       variant.Barcode,
+				Price:         convertToPrice(variant.Price),
+				CostPrice:     convertToPrice(variant.CostPrice),
+				ImageUrl:      imageUrl,
+				VatPercentage: zettleProduct.VatPercentage,
+				Status:        api.Status_ACTIVE,
+				UnitType:      convertToUnitType(zettleProduct.UnitName),
+				CreatedAt:     convertToTimestamp(zettleProduct.Created),
+				UpdatedAt:     convertToTimestamp(zettleProduct.Updated),
+			})
+		}
+	}
+
+	return products, nil
+}
+
+func convertToPrice(zettlePrice *zettle.Price) *api.Price {
+	if zettlePrice == nil {
+		return nil
+	}
+
+	return &api.Price{
+		Amount:     zettlePrice.Amount,
```

Comment:
I think we could do the conversion of the amount to a float here already, to simplify the logic for consumers.

Reply:
I don't agree. From what I read, currencies should be handles in their smallest unit, to avoid floating point numbers. 

See https://arc.net/l/quote/qigjaajz, https://archive.is/FkYvn.